DCS_RE_DLL/DCS_OPS_RE_Normal/dumps/JConfigurator.dll.txt
Microsoft (R) COFF/PE Dumper Version 14.51.36256.0
Copyright (C) Microsoft Corporation.  All rights reserved.


Dump of file G:\DCS_OPS_RE\bin\JConfigurator.dll

File Type: DLL

  Section contains the following exports for JConfigurator.dll

    00000000 characteristics
    FFFFFFFF time date stamp
        0.00 version
           1 ordinal base
          28 number of functions
          28 number of names

    ordinal hint RVA      name

          1    0 00007570 ?forceShutdown@jconf@@YAXXZ
          2    1 00007580 ?getDumps@jconf@@YA?AV?$basic_string@D@ed@@PEAPEBD@Z
          3    2 000075B0 ?getIntParameter@jconf@@YAHPEBDH@Z
          4    3 00007720 ?registerBackendStatCallback@jconf@@YAXP6A?AV?$basic_string@D@ed@@XZ@Z
          5    4 00007730 ?registerBool@jconf@@YAXPEBDV?$tuple@V?$function@$$A6A_NXZ@std@@V?$function@$$A6AX_N@Z@2@@std@@W4RunInThread@1@@Z
          6    5 000078F0 ?registerColor@jconf@@YAXPEBDAEAVVec3f@osg@@W4RunInThread@1@@Z
          7    6 00007940 ?registerColor@jconf@@YAXPEBDV?$tuple@V?$function@$$A6A?AVVec3f@osg@@XZ@std@@V?$function@$$A6AXAEBVVec3f@osg@@@Z@2@@std@@W4RunInThread@1@@Z
          8    7 00007B00 ?registerCommon@jconf@@YAXPEBDAEBV?$vector@V?$basic_string@D@ed@@V?$allocator@V?$basic_string@D@ed@@@2@@ed@@@Z
          9    8 00007C30 ?registerDouble@jconf@@YAXPEBDNNV?$tuple@V?$function@$$A6ANXZ@std@@V?$function@$$A6AXN@Z@2@@std@@W4RunInThread@1@@Z
         10    9 00007E10 ?registerDropDownList@jconf@@YAXPEBDAEBV?$vector@V?$basic_string@D@ed@@V?$allocator@V?$basic_string@D@ed@@@2@@ed@@V?$function@$$A6AXI@Z@std@@IW4RunInThread@1@@Z
         11    A 00008000 ?registerDropDownList@jconf@@YAXPEBDPEAPEBDIV?$function@$$A6AXI@Z@std@@IW4RunInThread@1@@Z
         12    B 00008320 ?registerDump@jconf@@YAXPEBDV?$function@$$A6AXAEAVJsonWriter@ed@@@Z@std@@@Z
         13    C 000083D0 ?registerEndpoint@jconf@@YAXPEBDV?$function@$$A6AXAEAVJsonWriter@ed@@@Z@std@@@Z
         14    D 00008480 ?registerFileSelector@jconf@@YAXPEBD0V?$function@$$A6AXPEBDPEBC_K@Z@std@@W4RunInThread@1@@Z
         15    E 00008600 ?registerFloat3@jconf@@YAXPEBDAEAVVec3f@osg@@W4RunInThread@1@@Z
         16    F 00008650 ?registerFloat3@jconf@@YAXPEBDV?$tuple@V?$function@$$A6A?AVVec3f@osg@@XZ@std@@V?$function@$$A6AXAEBVVec3f@osg@@@Z@2@@std@@W4RunInThread@1@@Z
         17   10 00008810 ?registerFloat@jconf@@YAXPEBDMMV?$tuple@V?$function@$$A6AMXZ@std@@V?$function@$$A6AXM@Z@2@@std@@W4RunInThread@1@@Z
         18   11 000089F0 ?registerFunc@jconf@@YAXPEBDV?$function@$$A6AXXZ@std@@W4RunInThread@1@@Z
         19   12 00008B60 ?registerInt@jconf@@YAXPEBDV?$tuple@V?$function@$$A6AHXZ@std@@V?$function@$$A6AXH@Z@2@@std@@W4RunInThread@1@@Z
         20   13 00008D20 ?registerStaticText@jconf@@YAXPEBDV?$function@$$A6A?AV?$basic_string@D@ed@@XZ@std@@@Z
         21   14 00008E90 ?registerTextEdit@jconf@@YAXPEBD0V?$function@$$A6AXPEBD@Z@std@@W4RunInThread@1@@Z
         22   15 000091F0 ?setIntParameter@jconf@@YAXPEBDH@Z
         23   16 00009390 ?setRenderThreadId@jconf@@YAXAEBVid@thread@ed@@@Z
         24   17 000091F0 ?toggleFunc@jconf@@YAXPEBD@Z
         25   18 00009630 ?unregister@jconf@@YAXPEBD@Z
         26   19 00009660 ?unregisterBackendStatCallback@jconf@@YAXXZ
         27   1A 00009670 ?unregisterDump@jconf@@YAXPEBD@Z
         28   1B 00009680 ?unregisterEndpoint@jconf@@YAXPEBD@Z

  Summary

        4000 .data
        3000 .pdata
       14000 .rdata
        1000 .reloc
        1000 .rsrc
       2B000 .text


DCS_RE_DLL/DCS_OPS_RE_Normal/demangled/JConfigurator.dll.txt
// DCS_OPS_RE: JConfigurator.dll
// Generated from dumpbin /exports

[    1]          0x0  void __cdecl jconf::forceShutdown(void)
[    2]          0x1  class ed::basic_string<char> __cdecl jconf::getDumps(char const **)
[    3]          0x2  int __cdecl jconf::getIntParameter(char const *, int)
[    4]          0x3  void __cdecl jconf::registerBackendStatCallback(class ed::basic_string<char> (__cdecl *)(void))
[    5]          0x4  void __cdecl jconf::registerBool(char const *, class std::tuple<class std::function<bool __cdecl(void)>, class std::function<void __cdecl(bool)>>, enum jconf::RunInThread)
[    6]          0x5  void __cdecl jconf::registerColor(char const *, class osg::Vec3f &, enum jconf::RunInThread)
[    7]          0x6  void __cdecl jconf::registerColor(char const *, class std::tuple<class std::function<class osg::Vec3f __cdecl(void)>, class std::function<void __cdecl(class osg::Vec3f const &)>>, enum jconf::RunInThread)
[    8]          0x7  void __cdecl jconf::registerCommon(char const *, class ed::vector<class ed::basic_string<char>, class ed::allocator<class ed::basic_string<char>>> const &)
[    9]          0x8  void __cdecl jconf::registerDouble(char const *, double, double, class std::tuple<class std::function<double __cdecl(void)>, class std::function<void __cdecl(double)>>, enum jconf::RunInThread)
[   10]          0x9  void __cdecl jconf::registerDropDownList(char const *, class ed::vector<class ed::basic_string<char>, class ed::allocator<class ed::basic_string<char>>> const &, class std::function<void __cdecl(unsigned int)>, unsigned int, enum jconf::RunInThread)
[   11]          0xA  void __cdecl jconf::registerDropDownList(char const *, char const **, unsigned int, class std::function<void __cdecl(unsigned int)>, unsigned int, enum jconf::RunInThread)
[   12]          0xB  void __cdecl jconf::registerDump(char const *, class std::function<void __cdecl(class ed::JsonWriter &)>)
[   13]          0xC  void __cdecl jconf::registerEndpoint(char const *, class std::function<void __cdecl(class ed::JsonWriter &)>)
[   14]          0xD  void __cdecl jconf::registerFileSelector(char const *, char const *, class std::function<void __cdecl(char const *, signed char const *, unsigned __int64)>, enum jconf::RunInThread)
[   15]          0xE  void __cdecl jconf::registerFloat3(char const *, class osg::Vec3f &, enum jconf::RunInThread)
[   16]          0xF  void __cdecl jconf::registerFloat3(char const *, class std::tuple<class std::function<class osg::Vec3f __cdecl(void)>, class std::function<void __cdecl(class osg::Vec3f const &)>>, enum jconf::RunInThread)
[   17]         0x10  void __cdecl jconf::registerFloat(char const *, float, float, class std::tuple<class std::function<float __cdecl(void)>, class std::function<void __cdecl(float)>>, enum jconf::RunInThread)
[   18]         0x11  void __cdecl jconf::registerFunc(char const *, class std::function<void __cdecl(void)>, enum jconf::RunInThread)
[   19]         0x12  void __cdecl jconf::registerInt(char const *, class std::tuple<class std::function<int __cdecl(void)>, class std::function<void __cdecl(int)>>, enum jconf::RunInThread)
[   20]         0x13  void __cdecl jconf::registerStaticText(char const *, class std::function<class ed::basic_string<char> __cdecl(void)>)
[   21]         0x14  void __cdecl jconf::registerTextEdit(char const *, char const *, class std::function<void __cdecl(char const *)>, enum jconf::RunInThread)
[   22]         0x15  void __cdecl jconf::setIntParameter(char const *, int)
[   23]         0x16  void __cdecl jconf::setRenderThreadId(class ed::thread::id const &)
[   24]         0x17  void __cdecl jconf::toggleFunc(char const *)
[   25]         0x18  void __cdecl jconf::unregister(char const *)
[   26]         0x19  void __cdecl jconf::unregisterBackendStatCallback(void)
[   27]         0x1A  void __cdecl jconf::unregisterDump(char const *)
[   28]         0x1B  void __cdecl jconf::unregisterEndpoint(char const *)


DCS_RE_DLL/DCS_OPS_RE_SCY/headers/Flight.dll/Params.hpp
#pragma once

// ============================================================
// AUTO-GENERATED RE HEADER
// ============================================================
// DLL: Flight.dll
// Class: Params
//
// WARNING:
// This is an export-derived RE interface.
// It is NOT a complete ABI reconstruction.
// Validate layout/vtables/calling conventions in Ghidra/x64dbg.
// ============================================================

namespace wGroundCrewOwner {
namespace ResourceProvider {

class Params
{
public:

    // RVA: 0x10 | Ordinal: 17
        void Params(struct wGroundCrewOwner::ResourceProvider::Params const &);

    // RVA: 0x11 | Ordinal: 18
        void Params(void);

    // RVA: 0x12 | Ordinal: 19
        void Params(struct wHeliport::ResourceProvider::Params &&);

    // RVA: 0x13 | Ordinal: 20
        void Params(struct wHeliport::ResourceProvider::Params const &);

    // RVA: 0x264 | Ordinal: 613
        void load(class Lua::Config &);

    // RVA: 0x267 | Ordinal: 616
        void load_maintenance_duration_(void);

    // RVA: 0x30 | Ordinal: 49
        void _Params(void);
};

} // namespace ResourceProvider
} // namespace wGroundCrewOwner

// DCS_OPS_RE_FLIGHT.DLL_PARAMS_HPP
